modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -192,6 +202,18 @@ public CompletableFuture<Void> startAsync(ComponentContext componentContext) {
             return nullCompletedFuture();
         }
 
+        CompletableFuture<Long> recoveryFinishFuture = metaStorageMgr.recoveryFinishedFuture();
+
+        assert recoveryFinishFuture.isDone();
+
+        long recoveryRevision = recoveryFinishFuture.join();
+
+        cleanUpResourcesForDroppedZonesOnRecovery();
+
+        CompletableFuture<Void> processZonesOnStart = processZonesOnStart(recoveryRevision, lowWatermark.getLowWatermark());
+
+        CompletableFuture<Void> processAssignmentsFuture = processAssignmentsOnRecovery(recoveryRevision);
```

Comment:
Is it intended not to chain processAssignmentsOnRecovery with processZonesOnStart?

Reply:
Chaining added.
`        CompletableFuture<Void> processZonesAndAssignmentsOnStart = processZonesOnStart(recoveryRevision, lowWatermark.getLowWatermark())
                .thenCompose(ignored -> processAssignmentsOnRecovery(recoveryRevision));`